HopsanGenerator/src/generators/HopsanGeneratorBase.cpp

```diff
@@ -930,6 +932,73 @@ bool HopsanGeneratorBase::generateCafFile(QString &rPath, ComponentAppearanceSpe
     return true;
 }
 
+bool HopsanGeneratorBase::generateComponentSourceFile(QString &path, ComponentSpecification &comp)
+{
+    QFile hppFile;
+    hppFile.setFileName(path);
+    if(!hppFile.open(QIODevice::WriteOnly | QIODevice::Text))
+    {
+        printErrorMessage("Could not open "+QFileInfo(path).absoluteFilePath()+" for writing.");
+        return false;
+    }
+
+    QTextStream out(&hppFile);
+    out << generateSourceCodefromComponentSpec(comp);
+
+    hppFile.close();
+    return true;
+}
+
+bool HopsanGeneratorBase::generateLibrarySourceFile(const ComponentLibrary &lib)
+{
```

Comment:
I think that this function duplicates some existing functionality inside the ComponentLibrary utility class. Check the "generateRegistrationCode" function

Reply:
Yes it does, but they have only two lines in common, and the utility function adds annoying comments about which library the component belongs to. These comments makes sense if we include a component in e.g. an FMU, but not in the source file for the actual library.

Reply:
The purpose of the comments was to make it easier for anyone who wants to edit the library manually after it has been created. But sure the number of comment can likely be reduced.
We can merge those two later, But I prefer to have one generator instead of two for the same thing.